pkg/pipelineascode/match.go

```diff
@@ -250,6 +258,19 @@ func filterRunningPipelineRunOnTargetTest(testPipeline string, prs []*tektonv1.P
 	return nil
 }
 
+func filterRunningPipelineRunForIncomingWebhook(eventType, incomingPipelineRun string, prs []*tektonv1.PipelineRun) []*tektonv1.PipelineRun {
+	if eventType != "incoming" {
+		return prs
+	}
+
+	for _, pr := range prs {
+		if incomingPipelineRun != "" && incomingPipelineRun == strings.TrimSuffix(pr.GetGenerateName(), "-") {
```

Comment:
what this do ? return only one incoming pr ? what about when you have multiple of them?

anyway this should go to the matcher...

Reply:
> what this do ? return only one incoming pr ? what about when you have multiple of them?

For incoming event even if there are multiple pipelineruns in .tekton folder but user will provide `pipelinerun` query parameter in curl request and that's  a mandatory parameter as per this  https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/adapter/incoming.go#L36 condition 

So there will be only one incoming PR at a time right ?
Or did i misunderstand this :thinking:

Reply:
i was the one who misunderstood it, you were right...